virttest/utils_test/qemu/__init__.py

```diff
@@ -188,45 +188,94 @@ def clear_win_driver_verifier(driver, vm, timeout=300):
 
 
 @error_context.context_aware
-def windrv_verify_running(session, test, driver, timeout=300):
+def _get_driver_id_once(session, driver_id_pattern, devcon_path):
+    """
+    Return the device hardware id of a specified device.
+
+    :param session: VM session
+    :param driver_id_pattern: Name pattern of the specified device
+    :param devcon_path: Full path for devcon.exe
+    """
+    driver_id_cmd = '%sdevcon.exe find * | findstr "VirtIO PVPanic"' % devcon_path
+    output = session.cmd_output(driver_id_cmd)
+    driver_id = re.findall(driver_id_pattern, output)
+    if driver_id:
+        driver_id = driver_id[0][::-1].split('\\', 1)[-1][::-1]
+    return driver_id
+
+
+@error_context.context_aware
+def get_driver_id(session, test, driver_id_pattern, devcon_path, timeout=300):
+    """
+    Return the device hardware id of a specified device in a period of time.
+
+    :param session: VM session
+    :param test: Kvm test object
+    :param driver_id_pattern: Name pattern of the specified device
+    :param devcon_path: Full path for devcon.exe
+    :param timeout: Timeout in seconds.
+    """
+    if not utils_misc.wait_for(lambda: _get_driver_id_once(session,
```

Comment:
Based on the docstring of  `utils_misc.wait_for`,

> If func() evaluates to True before timeout expires, return the
> value of func(). Otherwise return None.

You can write it like this:
```python
driver_id = utils_misc.wait_for(lambda: _get_driver_id_once(session,driver_id_pattern, devcon_path), timeout, 0, 5)
if driver_id:
     return driver_id
test.error("Didn't find driver info from guest")
```
The way you wrote it,  `_get_driver_id_once` will be called once more than it was needed.

And if `_get_driver_id_once` will never used in other functions,  you could put it into `get_driver_id`.